### PR TITLE
Don't zero out the z coordinate on load

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/Entities/EntityManager.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ namespace OpenLoco::EntityManager
         return removeFromSpatialIndex(entity, index);
     }
 
-    void moveSpatialEntry(EntityBase& entity, const Map::Pos2& loc)
+    void moveSpatialEntry(EntityBase& entity, const Map::Pos3& loc)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Entities/EntityManager.cpp:202]: Previously if the caller passed the loc by reference and the reference was the entities position you would end up zeroing out the z coordinate. Solution is to either not pass by reference or to always pass the full size of the coordinate despite only using the first 2 coordinates.

**Comment by AaronVanGeffen** [comment]
Does this solve the problem with vehicles appearing underground?

**Comment by duncanspumpkin** [comment]
Yes

**Comment by AaronVanGeffen** [comment]
The following save still presents the issue: [Desert_Delirium_1930d.SV5.zip](https://github.com/OpenLoco/OpenLoco/files/7164554/Desert_Delirium_1930d.SV5.zip)

Could it be that the save has been corrupted somehow? The problem sorts itself once trains leave the station.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**